batch_jobs/bucket_replicate/bucket_replicate_job.py

```diff
@@ -0,0 +1,145 @@
+    """
+    Start to run an job to generate bucket manifest
+    Args:
+        source_bucket(str): source bucket name
+        destination_bucket(str): destination bucket name
+        job_queue(str): job queue name
+        job_definition(str): job definition name
+
+    
+    Returns:
+        bool: True if the job was submitted successfully
+    """
+    keys = list_objects(source_bucket)
+    submit_jobs(source_bucket, destination_bucket, job_queue, job_definition, keys)
+
+
+def submit_job(source_bucket, destination_bucket, job_queue, job_definition, key):
+    """
+    Submit job to the job queue
+
+    Args:
+        source_bucket(str): source bucket
+        destination_bucket(str): destination bucket
+        job_queue(str): job queue name
+        job_definition(str): job definition name
+        key(str): S3 object key
+    
+    Returns:
+        bool: True if the job was submitted successfully
+    """
+    client = boto3.client("batch", region_name=REGION)
+    n_tries = 0
+
+    while n_tries < MAX_RETRIES:
+        try:
+            client.submit_job(
+                jobName="bucket_replicate",
+                jobQueue=job_queue,
+                jobDefinition=job_definition,
+                containerOverrides={"environment": [{"value": key, "name": "KEY"},{"value": source_bucket, "name": "SOURCE_BUCKET"},{"value": destination_bucket, "name": "DESTINATION_BUCKET"}]},
+            )
+            logging.info("submitting job to copy file {}".format(key))
+            return True
+        except ClientError as e:
+            if e.response["Error"]["Code"] == "AccessDeniedException":
+                logging.error(
+                    "ERROR: Access denied to {}. Detail {}".format(job_queue, e)
+                )
+                sys.exit(1)
+            if e.response["Error"]["Code"] != "TooManyRequestsException":
+                n_tries += 1
+                logging.info("{}. Retry {}".format(e, n_tries))
+            else:
+                logging.info("TooManyRequestsException. Sleep and retry...")
+
+        time.sleep(2 ** n_tries)
+    return False
+
+
+def submit_jobs(source_bucket, destination_bucket, job_queue, job_definition, keys):
+    """
+    Submit jobs to the queue
+
+    Args:
+        source_bucket(str): source bucket
+        destination_bucket(str): destination bucket
+        job_queue(str): job queue name
+        job_definition(str): job definition name
+        keys(list(str)): list of object keys
+    
+    Returns:
+        None
+    """
+    par_submit_job = partial(submit_job, source_bucket, destination_bucket, job_queue, job_definition)
+    with Pool(NUMBER_OF_THREADS) as pool:
+        pool.map(par_submit_job, keys)
+
+
+def list_objects(bucket_name):
+    """
+    List all objects in the bucket
+
+    Args:
+        bucket_name(str): the bucket name
+
+    Returns:
+        list(str): list of the objects
+    """
+    result = []
+
+    with open("/bucket-replicate/creds.json") as creds_file:
```

Comment:
```suggestion
aws_access_key_id = None
aws_secret_access_key = None
try:
    with open("/bucket-replicate/creds.json") as creds_file:
        creds = json.load(creds_file)
        aws_access_key_id = creds.get("aws_access_key_id")
        aws_secret_access_key = creds.get("aws_secret_access_key")
    
except IOError as e:
    logging.warn(f"Can not read /bucket-replicate/creds.json". Detail {str(e)})

```

Reply:
To list objects in given bucket, we can either grant access to the service account attached to the job or use k8s secret. The suggested code will support both approaches.